### PR TITLE
Fix tests on macOS 11

### PR DESCRIPTION
**Pull request by rokm**
This is a follow-up to discussion in #5464. There are three problematic tests that are failing on macOS Big Sur:
* test_depend_utils.py::test_ctypes_util_find_library
* test_apple_events.py::test_osx_custom_protocol_handler
* test_apple_events.py::test_osx_event_forwarding

The first one is failing because it tries to find system `c` library via `ctypes.util.find_library()`. As system libraries are hidden on Big Sur, this does not work - we now skip the test.

The other two tests use custom URL schema, and on Big Sur, this does not seem to work if the .app bundle is in the temporary directory (`/priv/var/...` or its symlink `/var/...`). Therefore, we now check the `tmpdir` prefix and xfail the test accordingly. 

For our macOS CI, we force `pytest` to use `basetemp` in user's home directory, so that those tests succeed.

The very first commit introduces `compat.is_macos_11`, which is used to identify Big Sur in the tests.

With these changes, we could switch/extend our macOS CI to Big Sur.

### CODE REVIEW

**Comment by rokm** [comment]
Here is the corresponding test run on Big Sur CI: https://github.com/rokm/pyinstaller/actions/runs/495836691

File: tests/unit/test_depend_utils.py [hunk 1]
```diff
@@ -58,6 +59,7 @@ def test_ctypes_LibraryLoader_LoadLibrary(monkeypatch, classname):
     assert res == set(['somelib.xxx'])
 
 
+@xfail(is_macos_11, reason='System libraries are hidden on macOS 11.')
```
Comment (BoboTiG) [inline tests/unit/test_depend_utils.py:62]: Maybe should we just `skip`?

**Review by BoboTiG — APPROVED** [review]

File: tests/unit/test_depend_utils.py (hunk 1 above)
Comment (rokm) [reply tests/unit/test_depend_utils.py:62]: Good point. I'll push a [skip ci] commit to change that once the current round of tests finishes.

**Review by bwoodsend — APPROVED** [review]

**Comment by bwoodsend** [comment]
@rokm I take it then that you can access a Big Sur when needed? Initially I thought @BoboTiG was our only Big Sur developer.

**Comment by BoboTiG** [comment]
Yes, I have access to Big Sur and Apple M1 machines.

**Comment by bwoodsend** [comment]
@BoboTiG Ooh, mind answering my questions over on #5315? The M1 users over there seem to have lost interest.

**Comment by rokm** [comment]
> @rokm I take it then that you can access a Big Sur when needed?

Yeah, I have access to Big Sur for testing now. Unfortunately it's not an M1 machine, though...

**Review by Legorooj — APPROVED** [review]
Looking good 👍

**Comment by rokm** [comment]
Superseded by #6260 and #6261.

**Pull request closed without merging**